examples/src/main/kotlin/io/ethers/examples/datadecoding/DataDecoding.kt

```diff
@@ -0,0 +1,56 @@
+package io.ethers.examples.datadecoding
+
+import io.ethers.abi.AbiCodec
+import io.ethers.abi.AbiFunction
+import io.ethers.core.types.Bytes
+import io.ethers.examples.gen.UniswapV2Router02
+import kotlinx.cli.ArgParser
+import kotlinx.cli.ArgType
+import kotlinx.cli.default
+
+/**
+ * Decoding HEX data from a given swapExactTokensForTokens transaction. This can be accomplished in two scenarios:
+ * when the ABI of the smart contract is available or when only the function signature is known.
+ */
+class DataDecoding(txInput: String) {
+    private val calldata = Bytes(txInput)
+    fun run() {
+        // # Decode function input data when ABI is known
+        val autoDec = UniswapV2Router02.FUNCTION_SWAP_EXACT_TOKENS_FOR_TOKENS.decodeCall(calldata)
+
+        // # Manually decode function input data when we only know function signature
+        val function = AbiFunction.parseSignature("swapExactTokensForTokens(uint256,uint256,address[],address,uint256)")
+        val manualDec = AbiCodec.decodeWithPrefix(4, function.inputs, calldata.value)
```

Comment:
This should be replaced with `function.decodeCall`. Normally, users should not use `AbiCodec` directly.